### DISCUSSION TITLE
Feature Wish List

### DISCUSSION THREAD

**Discussion by ejoerns** [rauc/rauc-hawkbit-updater#108] [Ideas]
Note: This was in the Wiki before as 'Roadmap'. Moved it to an issue to close the (unused) wiki but not lose these items.

- [ ] "Ask" permission to download (D-BUS).
- [ ] "Ask" permission to install (D-BUS).
- [ ] Expose download and install progress via D-BUS (like Rauc service does).
- [ ] Support casync

**Comment by jluebbe** [comment]
> Note: This was in the Wiki before as 'Roadmap'. Moved it to an issue to close the (unused) wiki but not lose these items.
> 
> * [ ]  "Ask" permission to download (D-BUS).
> * [ ]  "Ask" permission to install (D-BUS).

Why should this be in rauc-hawkbit-updater instead of rauc itself?

> * [ ]  Expose download and install progress via D-BUS (like Rauc service does).
> * [ ]  Support casync

Isn't this better handled by using RAUC's streaming mode? I wouldn't like to just wrap the existing rauc D-Bus API.

**Comment by ejoerns** [comment]
> > Note: This was in the Wiki before as 'Roadmap'. Moved it to an issue to close the (unused) wiki but not lose these items.
> > 
> > * [ ]   "Ask" permission to download (D-BUS).
> > * [ ]   "Ask" permission to install (D-BUS).
> 
> Why should this be in rauc-hawkbit-updater instead of rauc itself?

The real download of a bundle (assuming not using streaming support) cannot be handled by RAUC as it happens before RAUC gets actually called. Asking permission for installation instead could be handled by RAUC.
A possible drawback could be that Hawkbit would assume the installation is started and in progress as long as the update was not confirmed (assuming physical interaction e.g.). Maybe this is acceptable/intended.

> > * [ ]   Expose download and install progress via D-BUS (like Rauc service does).
> > * [ ]   Support casync
> 
> Isn't this better handled by using RAUC's streaming mode? I wouldn't like to just wrap the existing rauc D-Bus API.

When being merged, RAUC streaming mode is possibly preferred, yes (despite it still lacks some additional features). But that was not available at the time of writing this list.
The comment about wrapping the D-Bus API targets the first item? Or how does this deal with casync?

**Comment by caglarkilimci** [comment]
Hello all,

These are really good and reasonable features. We exactly consider these features before seeing this page. Is there any plan to do these? I can also help to implement.

Sincerely,
Caglar

**Comment by Bastian-Krause** [comment]
> Note: This was in the Wiki before as 'Roadmap'. Moved it to an issue to close the (unused) wiki but not lose these items.
> 
> * [ ]  "Ask" permission to download (D-BUS).
> * [ ]  "Ask" permission to install (D-BUS).

hawkBit implemented user confirmation in its newest release. See also #156.
